Support  laravel 10 and 11

ahmdadl opened the pull request:
Support laravel version 10 and above

ahmdadl committed:
update: laravel +10

ahmdadl committed:
downgrade mongo to 4 as from 5 an up id is reserved and can not be used

ahmdadl committed:
fix: request user not working in queued jobs

ahmdadl committed:
update Readme.md

ahmdadl committed:
support laravel 11

ahmdadl committed:
support laravel 11

ahmdadl committed:
support laravel 11

ahmdadl committed:
support laravel 11

ahmdadl committed:
support laravel 11

ahmdadl committed:
fix: remove unsupported test methods

ahmdadl committed:
feat: enhance types

hassanzohdy commented on src/Helpers/functions.php:
```diff
@@ -105,10 +105,10 @@ function pred(...$vars)
      * 
      * @param string $repository
-     * @return \HZ\Illuminate\Mongez\Repository\RepositoryInterface
+     * @return \HZ\Illuminate\Mongez\Repository\MongoDBRepositoryManager
```
It should be the repository interface not mongodb so it could be working on both MYSQL and MongoDB

ahmdadl committed:
feat: enhance types

ahmdadl commented:
Updated to use RepositoryInterface and generics for proper method autocompletion

- Implemented RepositoryInterface to ensure type safety and consistency.
- Utilized generics to allow IDEs to infer and autocomplete the actual repository class methods.
- Improved code readability and developer experience with enhanced autocompletion.
